### PR TITLE
refactor: rename custom errors

### PR DESCRIPTION
**Pull request by binbin-li**
### What?

One of PR series to address: https://github.com/notaryproject/notation/issues/287

1. Rename custom errors to make errors can be grouped together by the prefix in go doc https://pkg.go.dev/github.com/notaryproject/notation-core-go@v0.0.0-20220803031124-6670fef9acec/signer
2. Add missing periods to comments.
3. Fix Verify logic to return payload and use the correct method.

Signed-off-by: Binbin Li <libinbin@microsoft.com>

### CODE REVIEW

**Review by patrickzheng200 — COMMENTED** [review]
lgtm

File: signature/algorithm.go [hunk 1]
```diff
@@ -57,8 +57,7 @@ func ExtractKeySpec(signingCert *x509.Certificate) (KeySpec, error) {
 			}, nil
 		default:
 			return KeySpec{}, UnsupportedSigningKeyError{
-				keyType:   KeyTypeRSA,
-				keyLength: key.Size(),
+				fmt.Sprintf("rsa algorithm of size %d is not supported", key.Size()),
```
Comment (shizhMSFT) [inline signature/algorithm.go:60]: Suggesting:
```go
// ExtractKeySpec extracts keySpec from the signing certificate
func ExtractKeySpec(signingCert *x509.Certificate) (KeySpec, error) {
	switch key := signingCert.PublicKey.(type) {
	case *rsa.PublicKey:
		switch bitSize := key.Size() << 3 {
		case 2048, 3072, 4096:
			return KeySpec{
				Type: KeyTypeRSA,
				Size: bitSize,
			}, nil
		default:
			return KeySpec{}, UnsupportedSigningKeyError{
				fmt.Sprintf("rsa key size %d is not supported", bitSize),
			}
		}
	case *ecdsa.PublicKey:
		switch bitSize := key.Curve.Params().BitSize {
		case 256, 384, 521:
			return KeySpec{
				Type: KeyTypeEC,
				Size: bitSize,
			}, nil
		default:
			return KeySpec{}, UnsupportedSigningKeyError{
				fmt.Sprintf("ecdsa key size %d is not supported", bitSize),
			}
		}
	}
	return KeySpec{}, fmt.Errorf("invalid public key type")
}
```

Comment (binbin-li) [reply signature/algorithm.go]: Good call, updated.

File: signature/errors.go [hunk 2]
```diff
@@ -4,48 +4,49 @@ import "fmt"
...
 	}
 	return "signing key is not supported"
 }
 
 // MalformedArgumentError is used when an argument to a function is malformed.
 type MalformedArgumentError struct {
-	param string
-	err   error
+	Param string
+	Err   error
```
[the remark this answers is not in the document]
Comment (binbin-li) [reply signature/errors.go:34]: Gotcha, added it.

File: signature/internal/base/envelope.go [hunk 3]
```diff
@@ -33,34 +33,36 @@ func (e *Envelope) Sign(req *signature.SignRequest) ([]byte, error) {
...
 	}
 
-	if _, _, err := e.Envelope.Verify(); err != nil {
+	payload, _, err := e.Envelope.Verify()
+	if err != nil {
 		return nil, nil, err
 	}
-	signerInfo, err := e.Envelope.SignerInfo()
+
+	signerInfo, err := e.SignerInfo()
```
[the remark this answers is not in the document]
Comment (binbin-li) [reply signature/internal/base/envelope.go:49]: gotcha

File: signature/errors.go [hunk 4]
```diff
@@ -4,48 +4,49 @@ import "fmt"
...
+}
+
+// NewMalformedArgumentError creates a new MalformedArgumentError with malformed
+// paramaters and the actual error
+func NewMalformedArgumentError(param string, err error) *MalformedArgumentError {
+	return &MalformedArgumentError{
+		Param: param,
+		Err: err,
+	}
 }
```
[the remark this answers is not in the document]
Comment (binbin-li) [reply signature/errors.go]: Removed

File: signature/internal/base/envelope.go [hunk 5]
```diff
@@ -10,14 +10,14 @@ import (
...
-	signature.Envelope
-	Raw []byte
+	signature.Envelope        // internal envelope in a specific format(COSE/JWS)
+	Raw                []byte // raw signature
 }
 
 var errorFunc = func(s string) error {
-	return signature.NewMalformedSignatureError(s)
+	return &signature.MalformedSignatureError{Msg: s}
 }
```
[the remark this answers is not in the document]
Comment (binbin-li) [reply signature/internal/base/envelope.go]: Removed

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**